### ISSUE TITLE
Support Vault credential at Ad Hoc command

### ISSUE THREAD

**Issue by CWollinger** [ansible/awx#5855]
##### ISSUE TYPE
 - Feature Idea

##### SUMMARY
In the inventory there are different usernames and passwords stored encrypted for the connections. In the job template a vault credential is provided to decrypt at the job execution. This is not possible for Ad Hoc command. You can only use credential from the kind `Machine`. 

It would be good to use other credentials types e.g. `Vault` for the Ad Hoc commands, if no key authentication is possible. Command for CLI would be `ansible -m command all -a 'ls /root' --ask-vault-pass`

**Comment by wenottingham** [comment]
~There's no vault passed to ad-hoc commands, so why would the credential be useful?~

**Comment by wenottingham** [comment]
This is solely for vault-encrypted variables in inventory?

**Comment by CWollinger** [comment]
Yes, e.g. if you use encrypted `ansible_user` or `ansible_password` vars from inventory.

**Comment by azrdev** [comment]
Defining one use case: Get to know which built-in/special variables are defined for a host. The documentation doesn't list them ( https://github.com/ansible/ansible/issues/25923 / https://github.com/ansible/ansible/issues/23572 ), firing a `setup` task gathers facts and displays only it, so firing a `debug var=vars` task would be the obvious workaround (since ad-hoc tasks don't gather facts).  
As soon as the inventory contains any vaults, this is not feasible anymore, but requires a dedicated Job Template (+ Project)

**Comment by jon4hz** [comment]
This would actually be very helpful. It's perfectly valid to use variables inside ad hoc commands which can be encrypted.

e.g. `ansible -m shell -a "echo {{ my_encrypted_var }}"`
